solutions/advanced/cses-1655.mdx

```diff
@@ -0,0 +1,92 @@
+---
+id: cses-1655
+source: CSES
+title: Maximum Xor Subarray
+author: Mihnea Brebenel
+---
+
+## Explanation
+
+Let $\texttt{pref}_i$ be the xor-sum of the prefix ending at $i$. Then, we must find a previous prefix sum
+$\texttt{pref}_j$ such that $\texttt{pref}_i \oplus \texttt{pref}_j$ is maximized. To obtain the maximal result
+the $0$ bits should be flipped to $1$. And this is where a trie data structure comes in handy. It can be used
+to maintain the binary representations of the xor-sums of prefixes. We add the values in trie from the most significant
+bit to the least significat bit. When querying for $\texttt{pref}_j$ we'll traverse the trie trying to down the path
+of the opposite bit, such that after xoring with $\texttt{pref}_i$ more bits are flipped to $1$.
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N \cdot \log{10^9})$
```

Comment:
replace 10^9 with a variable and explain what that variable is

Reply:
oki doki